docs/source/user-guide/example-usage.md

```diff
@@ -81,6 +81,26 @@ async fn main() -> datafusion::error::Result<()> {
 +---+--------+
 ```
 
+## Arrow Versions
+
+Many of DataFusion's public APIs use types from the
+[`arrow`](https://docs.rs/arrow/latest/arrow/) crate, so if you use
+`arrow` in your project, its version must match that used by
+DataFusion.
+
+The easiest way to ensure the versions match is to use the `arrow`
+exported by DataFusion, for example:
```

Comment:
It may be good to explain which crate versions cause this issue. When you wrote, ` its version must match that used by
DataFusion.` this is indeed correct. But most people may check out the arrow crate still and go for 40.0, instead of the 39.0 version that is used by Datafusion. Would it be okay to add more transparency or example cases that would cause that mismatched types to happen? (Which using Datafusion crate with 40.0 arrow would cause this intermingle behavior).

Reply:
Granted the arrow dependency to use for the Datafusion crate is stated in the crates.io. This may just be a little nit picky and not needed.

Reply:
Thanks! I tried to clarify in e420deb04 -- let me know how if that makes it clearer.

Reply:
Looks great. LGTM. Thank you for doing this.